gems/pending/Scvmm/miq_hyperv_disk.rb

```diff
@@ -0,0 +1,171 @@
+$LOAD_PATH.push("#{File.dirname(__FILE__)}/../util")
+
+require 'miq_winrm'
+require 'miq_scvmm_parse_powershell'
+require 'base64'
+require 'securerandom'
+require 'memory_buffer'
+
+require 'rufus/lru'
+
+class MiqHyperVDisk
+  MIN_SECTORS_TO_CACHE = 8
+  DEF_BLOCK_CACHE_SIZE = 300
+  DEBUG_CACHE_STATS    = false
+
+  attr_reader :hostname, :virtual_disk, :file_offset, :file_size, :parser, :vm_name, :temp_snapshot_name
+
+  def initialize(hyperv_host, user, pass, port = nil)
+    @hostname  = hyperv_host
+    @winrm     = MiqWinRM.new
+    port ||= 5985
+    options = {:port => port, :user => user, :pass => pass, :hostname => @hostname}
+    @connection  = @winrm.connect(options)
+    @parser      = MiqScvmmParsePowershell.new
+    @block_size  = 4096
+    @file_size   = 0
+    @block_cache = LruHash.new(DEF_BLOCK_CACHE_SIZE)
+    @cache_hits  = {}
+    @cache_misses = {}
```

Comment:
Sorry, I wasn't clear. You can set a default value for a hash when it's instantiated. The previous 2 lines should be:
```
@cache_hits  = Hash.new(0)
@cache_misses = Hash.new(0)
```
Then, the initialization code in `open` can be removed.